torch/backends/xeon/run_cpu.py

```diff
@@ -474,94 +488,117 @@ def launch(self, args):
             if (
                 not args.multi_instance
                 and args.ninstances == -1
-                and args.ncores_per_instance == -1
+                and args.ncores_per_instance == [-1]
             ):
                 args.ninstances = 1
-                args.ncores_per_instance = len(cores)
+                args.ncores_per_instance = [len(cores)]
             elif (
                 args.multi_instance
                 and args.ninstances == -1
-                and args.ncores_per_instance == -1
+                and args.ncores_per_instance == [-1]
             ):
                 args.throughput_mode = True
-            elif args.ncores_per_instance == -1 and args.ninstances != -1:
+            elif args.ncores_per_instance == [-1] and args.ninstances != -1:
                 if args.ninstances > len(cores):
                     raise RuntimeError(
                         f"there are {len(cores)} total cores but you specify {args.ninstances} ninstances; \
 please make sure ninstances <= total_cores)"
                     )
+                if args.skip_cross_node_cores:
```

Comment:
Maybe `skip_cross_node_instances` is better since cores cannot be cross node but instances can.